### DISCUSSION TITLE
Few questions

### DISCUSSION THREAD

**Discussion by JaineshPathak** [jelte/be.khepri.play.assetdelivery.addressables#45] [Q&A]
I am working in a small team and currently using Addressables 1.8.5 in Unity 2019.3.15f1.
Currently we are using Google Cloud Storage as CDN and we are finding it costly. As a result, we think using PAD is a good option.

1. I just have to add Play Asset Delivery schema to existing Addressable groups and that's it?
2. Also do I have to change the BuildPath and LoadPath to "LocalBuildPath" and "LocalLoadPath"? I was just reading this tutorial: https://killertee.wordpress.com/2021/09/04/unity-play-asset-delivery-and-addressable-assets/
3. I have heard that using pure PAD without Addressables cannot contain executable scripts? Will this be same problem with your PAD + Addressables plugin? The thing is I am using some scripts which are directly embedded inside an asset bundle.
4. When building Asset bundles, will it work when I choose "Update Existing Build" in Addressables Groups window? Or do I have to make a completely new fresh build which I am frightened about.

**Comment by jelte** [comment]
Yes, to reduce CDN cost, PAD would definitly be a good option, just know that there are limitations to it in terms of max size and number of bundles (see the PAD documentation).

1. Yes, that's about all you need to do, (and follow the build steps of course)
2. Yes, Unity changed those "recently", as such older projects are not using the same Local.BuildPath and Local.LoadPath  as in `AddressableAssetSettings.kLocalBuildPath`
3. Not sure, never done this. But if PAD doesn't support it, this won't either. C# scripts need to be compiled, I don't think unity support hotloading code ? but I could be mistaken.
4. as you are changing config of you bundles, it is probably safer to just do a fresh build. But test it out first and see if update works.

Sorry that most of the answers are "try it out", but each usecase is different, and I haven't had a need to use PAD since I made this package, so my own usage of this is limited to the basics.

**Reply by JaineshPathak** [reply]
Thanks for the quick reply.

I have another question.
So say I update something in any Asset bundle like changing images, sounds, etc.

I have to repeat these steps every time right?

1. Build the Asset Bundles like normal from Addressable Groups window
2. Click on Google > Create config for Addressables Groups
3. Click on Google > Build Android App Bundle (.aab)
4. Upload the new .aab file to Play Store and new version will download new updated bundles right?

**Reply by jelte** [reply]
pretty much, yes.
